tracing/tracing.go

```diff
@@ -6,51 +6,194 @@ package tracing
+			if value, err := strconv.ParseInt(e, 10, 0); err == nil {
+				cfg.port = int(value)
+			} else {
+				return nil, errors.Wrapf(err, "cannot parse env var %s=%s", envJaegerAgentPort, e)
+			}
+		}
+	}
 
-	// put the metricsFactory earlier so provided options can override it
-	opts := append([]jaegercfg.Option{jaegercfg.Metrics(metricsFactory)}, options...)
+	// Then we parse the sampler type, if it is not exist then we default to const
+	if e := os.Getenv(envSamplerType); e != "" {
+		cfg.samplerType = e
+	}
+
+	if e := os.Getenv(envSamplerParam); e != "" {
+		if value, err := strconv.ParseFloat(e, 64); err == nil {
+			cfg.samplerParam = value
+		} else {
+			return nil, errors.Wrapf(err, "cannot parse env var %s=%s", envSamplerParam, e)
+		}
+	}
+
+	if e := os.Getenv(envSamplerRemoteURL); e != "" {
+		cfg.samplerRemoteURL = e
+	} else if e := os.Getenv(envJaegerAgentHost); e != "" {
+		// Fallback if we know the agent host - try the sampling endpoint there
+		cfg.samplerRemoteURL = fmt.Sprintf("http://%s:%d/sampling", e, DefaultSamplingServerPort)
+	}
 
-	closer, err := cfg.InitGlobalTracer(serviceName, opts...)
+	cfg.customAttributes, err = parseAttributes(os.Getenv("JAEGER_TAGS"))
 	if err != nil {
-		return nil, errors.Wrap(err, "could not initialize jaeger tracer")
+		return nil, errors.Wrap(err, "could not parse JAEGER_TAGS")
 	}
-	return closer, nil
+	return &cfg, nil
 }
 
-// NewFromEnv is a convenience function to allow tracing configuration
-// via environment variables
-//
-// Tracing will be enabled if one (or more) of the following environment variables is used to configure trace reporting:
-// - JAEGER_AGENT_HOST
-// - JAEGER_SAMPLER_MANAGER_HOST_PORT
-func NewFromEnv(serviceName string, options ...jaegercfg.Option) (io.Closer, error) {
-	cfg, err := jaegercfg.FromEnv()
+func (cfg config) initJaegerTracerProvider(serviceName string) (*Closer, error) {
+	// Read environment variables to configure Jaeger
+	var ep jaegerotel.EndpointOption
+	// Create the jaeger exporter: address can be either agent address (host:port) or collector URL
+	if cfg.host != "" {
+		ep = jaegerotel.WithAgentEndpoint(jaegerotel.WithAgentHost(cfg.host), jaegerotel.WithAgentPort(strconv.Itoa(cfg.port)))
+	} else {
+		ep = jaegerotel.WithCollectorEndpoint(jaegerotel.WithEndpoint(cfg.collectorURL))
+	}
+	exp, err := jaegerotel.New(ep)
 	if err != nil {
-		return nil, errors.Wrap(err, "could not load jaeger tracer configuration")
+		return nil, err
 	}
 
-	if cfg.Sampler.SamplingServerURL == "" && cfg.Reporter.LocalAgentHostPort == "" && cfg.Reporter.CollectorEndpoint == "" {
-		return nil, ErrBlankTraceConfiguration
+	res, err := NewResource(serviceName, cfg.customAttributes)
+	if err != nil {
+		return nil, err
+	}
+
+	// Configure sampling strategy
+	sampler := tracesdk.AlwaysSample()
+	if cfg.samplerType == "const" || cfg.samplerType == "probabilistic" {
+		sampler = tracesdk.TraceIDRatioBased(cfg.samplerParam)
```

Comment:
Idea: how about adding this as a resource attribute? I don't think we store it by default in OTel, but we did have it in the Jaeger client.

Reply:
Hello @jpkrohling, thanks for the input and would like to engage in a discussion to gain a deeper understanding of your perspective. However you are off this week, I reached out to @gouthamve instead. Upon conversing with him, we concluded that the current situation lacks an optimal solution. Setting it as a resource attribute would lead to a static value if not restart, out of sync with any updates to the remote sampling setting.

Given these circumstances, it appears prudent to put this matter on hold for now. I suggest revisiting it when the community arrives at a more suitable solution. Your insights remain valuable, and I look forward to further discussions.